src/inbox/mod_inbox_rdbms.erl

```diff
@@ -61,6 +63,17 @@ get_inbox_rdbms(LUser, LServer, #{ order := Order } = Params) ->
                  " ORDER BY timestamp ", OrderSQL, ";"],
     mongoose_rdbms:sql_query(LServer, Query).
 
+
+get_inbox_unread(Username, Server) ->
+    Res = mongoose_rdbms:sql_query(Server,
+                                   ["select unread_count from inbox "
+                                    "WHERE luser=", esc_string(Username),
+                                    ";"]),
+    {ok, Val} = check_result(Res),
+    %% We read unread_count value when the message is sent and is not yet in receiver inbox
```

Comment:
Is it because of the hook's handler ordering? I'd check if we can change the order how hook handlers are run (push notification's handler run after inbox's handler).

Reply:
Push notification's handler is run after inbox's handler in case of `user_send_packet` hook. In case of `filter_packet` the push handler is run first. Otherwise in case of private messages the returned value is not yet incremented, but in groupchat it is. I didn't know how to handle it differently

Reply:
It still seams to me that `filter_packet` handler in `mod_inbox` can be run before `filter_packet` in `mod_push` (or whatever is the full correct name for this module :) ). Or does it cause other issues?